src/useWindowScroll.ts

```diff
@@ -22,6 +22,10 @@ const useWindowScroll = (): State => {
       });
     };
 
+    //We have to update window scroll at mount, before subscription.
+    //Window scroll may be changed between render and effect handler.
+    handler();
```

Comment:
I find that all tests still pass if I comment out this line, so it appears that the edge case you're fixing is not tested properly.

Reply:
Here's a test that I came up with, which will fail if this call is not in place :-)

```typescript
it('should set window scroll if changed after mounting', () => {
  setWindowScroll(1, 2);

  const TestComponent = () => {
    useEffect(() => {
      // Simulate window scroll changing between after render,
      // but before adding the event listener
      setWindowScroll(2, 3);
    });

    const { x, y } = useWindowScroll();
    return (
      <p>
        {x},{y}
      </p>
    );
  };

  render(<TestComponent />);

  reactDomAct(() => {
    requestAnimationFrame.step();
  });

  screen.getByText('2,3');
});
```

Note that I have to use `act` from `react-dom/test-utils'` which is different than the one from `@testing-library/react-hooks`, otherwise the test fails. I have imported it with:

```javascript
import { act as reactDomAct } from 'react-dom/test-utils';
```

Reply:
fixed here: https://github.com/streamich/react-use/pull/1700/commits/92f3fa7599e5629981e5b7a6b8b9d021f99d3114
I dont neeed act react-dom/test-utils, i use it from @testing-library/react, with render from @testing-library/react